allennlp/service/server_sanic.py

```diff
@@ -135,4 +193,18 @@ def _caching_prediction(model: Predictor, data: str) -> JsonDict:
         """list the available models"""
         return response.json({"models": list(app.predictors.keys())})
 
+    # As a SPA, we need to return index.html for /model-name and /model-name/permalink
```

Comment:
@codeviking I remember you cautioning against SPAs at one point.  Do any of your concerns apply here?

Reply:
Your demo is a bit closer to something that's SPA appropriate.  S2 less so, y'all are good to go :)